sygnal/apnspushkin.py

```diff
@@ -79,9 +79,21 @@ class ApnsPushkin(ConcurrencyLimitedPushkin):
     Relays notifications to the Apple Push Notification Service.
     """
 
-    # Errors for which the token should be rejected
-    TOKEN_ERROR_REASON = "Unregistered"
-    TOKEN_ERROR_CODE = 410
+    # Errors for which the token should be rejected and not reused
+    # See https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns  # noqa: E501
+    # for the full list of possible errors.
+    TOKEN_ERRORS = {
+        # A client has uploaded an invalid token.
+        (400, "BadDeviceToken"),
+        # `DeviceTokenNotForTopic` may be due to a bad token or a Sygnal
+        # misconfiguration. In the event of a misconfiguration, clients will need to
+        # reupload their tokens to their homeserver. Element iOS and clients based on
+        # matrix-ios-kit already do this periodically.
```

Comment:
Sygnal is supposed to be agnostic to Element iOS, but this seems to give good guidance here or what's going on.

I think matrix-ios-kit is also being deprecated?

```suggestion
        # reupload their tokens to their homeserver.
```